src/tests.c

```diff
@@ -438,6 +438,29 @@ void run_sha256_tests(void) {
     }
 }
 
+/* Tests for the equality of two sha256 structs. This function only produces a
+ * correct result if an integer multiple of 64 many bytes have been written
+ * into the hash functions. */
+void test_sha256_eq(secp256k1_sha256 *sha1, secp256k1_sha256 *sha2) {
+    unsigned char buf[32] = { 0 };
+    unsigned char buf2[32];
+
+    /* Is buffer fully consumed? */
+    CHECK((sha1->bytes & 0x3F) == 0);
```

Comment:
Should we check `sha2` as well? Also should this be a `VERIFY_CHECK`? I guess in tests.c the distinction doesn't really matter.

Reply:
Tests should use CHECK, because its useful to run the tests without -DVERIFY to make sure that the verifys aren't fixing a bug (e.g. with a side effect in a verify macro)-- and you don't want all the tests to go away when you do that. :)  There is currently no VERIFY_CHECK in tests.c